### DISCUSSION TITLE
Typing supabase severless functions or client.invoke?

### DISCUSSION THREAD

**Discussion by ChuckJonas** [supabase/supabase#13621] [Questions]
The docs have a section for [typescript support ](https://supabase.com/docs/guides/functions/typescript-support), but don't actually mention anything about typing request to invoke.  

The `supabase gen types typescript --local > src/database.types.ts` doesn't generate database types for the functions... which probably makes sense, because they are typescript already.

But I'm wonder what the best approach is for having type safety between the severless function and `client.invoke`?  I'm guessing it would require exporting some sort of type and then using a higher order function to enforce that the serverless function adheres to it?

Looking at the generic signature, it actually doesn't support typing the request type, which means in the current state, I guess the best you get is having the response type lineup...

**Comment by matart15** [comment]
Yes having same problem
